wgpu-hal/src/vulkan/instance.rs

```diff
@@ -557,6 +560,21 @@ impl crate::Instance<super::Api> for super::Instance {
             layers
         };
 
+        #[cfg(target_os = "android")]
+        let android_sdk_version = {
+            // See: https://developer.android.com/reference/android/os/Build.VERSION_CODES
+            let mut prop = android_properties::getprop("ro.build.version.sdk");
+            if let Some(val) = prop.value() {
+                u32::from_str_radix(&val, 10)
+                    .expect("Failed to parse ro.build.version.sdk property")
```

Comment:
Could this also log on failure, printing the value it tried and failed to parse?

Reply:
right, yeah I just realized I missed this, by bad - I'll update and also use `val.parse::<u32>()`  as suggested by clippy